mantidproject/mantid
# Engineering Diffraction GUI integrate multi-run widget into GSAS tab

The new multi-run widget has been integrated with the GSAS tab of the ED GUI. This widget contains focused runs and their fitted peaks workspaces, indexed by run number and bank ID. Workspaces can be plotted, with or without their fitted peaks if available, removed from the widget, and plotted to a separate window. This replicates the behaviour of the fitting tab - the aim is to replace the relevant section of the fitting tab with this widget as well.

**To test:**

Sample data files [here](https://github.com/mantidproject/mantid/files/1652777/engggui_pr_data.zip). 

Load up the ED GUI (`Interfaces -> Diffraction -> Engineering Diffraction`) then click `GSAS Fitting` tab.

Click the first **Browse** button to browse for focused run data (in the zip file). Load all 3 runs, and make sure `277208_1`, `277208_2` and `277209_2` are displayed in the list widget next to the plot. Clicking each of them in turn should plot the run. Try only loading 2 runs, and make sure the third is added correctly once loaded on its own (you'll need to exit the GUI to get rid of the loaded runs).

Make sure the zoom tool works nicely as well - zoom by clicking and dragging, and reset by right-clicking

Try removing runs and plotting to a separate window. Note that, since GSASIIRefineFitPeaks isn't yet hooked up to the GUI, plotting fitted peaks is not yet supported.

Partially addresses #21640 

**Release Notes** 
Internal change, doesn't need to be in the release notes

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
